### PR TITLE
Generate project structure with global.json

### PR DESCRIPTION
**Pull request by david-driscoll**
This adds support for the discussion in #367

yo-project

### CODE REVIEW

**Comment by sayedihashimi** [comment]
Beta8 is getting released tomorrow so I'm inclined to release that, merge this after, test it and then rerelease. Any concerns with that?

**Comment by spboyer** [comment]
Agreed in releasing beta8. Then I'd like to have some more discussion around the "project" structure PR. Is this something That we are looking to force or as an option? 

.sln files are only required for VS and even asking for them in a non VS space seems intrusive, maybe the wrong word. VS will create the files it needs.

One of the major selling points of the new project structure was that it was file/folder based. This makes it feel like. Step back. My 2 cents. Maybe it's a nickel...

**Comment by david-driscoll** [comment]
Well it doesn't include an actual solution file, instead it's just
following the structure for the the project locations.

i.e. /src, /tests, /global.json, etc.

The when you add a new project using yeoman to a project with the same
structure, well add the project into the appropriate folder. As it is today
it's easy to create a simple single project, but expanding to multiple is
comber some and frankly frustrating.

It may be that the terminology needs to change, that's just a game of
finding the right words which as we all know is one of the core problems in
programming :)

On Wed, Oct 14, 2015 at 6:24 PM Shayne Boyer <notifications@github.com>
wrote:

> Agreed in releasing beta8. Then I'd like to have some more discussion
> around the "project" structure PR. Is this something That we are looking to
> force or as an option?
>
> .sln files are only required for VS and even asking for them in a non VS
> space seems intrusive, maybe the wrong word. VS will create the files it
> needs.
>
> One of the major selling points of the new project structure was that it
> was file/folder based. This makes it feel like. Step back. My 2 cents.
> Maybe it's a nickel...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/generator-aspnet/pull/418#issuecomment-148220544>
> .
>
-- 
Thanks,
David

**Comment by david-driscoll** [comment]
@spboyer sorry was at a meetup last night when I replied.  Yeah, maybe "make solution file" is bad wording (well okay, it is bad wording, lol).

Perhaps `Create solution structure?` would be better?

**Comment by david-driscoll** [comment]
I've updated for beta8

File: app/index.js [hunk 1]
```diff
@@ -99,147 +155,230 @@ var AspnetGenerator = yeoman.generators.Base.extend({
...
+
+      done();
+      return;
+    }
+
+    var prompts = [{
+      type: 'confirm',
+      name: 'projectStructure',
+      message: 'Create a solution file?',
+      default: true
```
Comment (peterblazejewicz) [inline app/index.js]: @david-driscoll 
I think the default should be `false`
Thoughts?

**Comment by peterblazejewicz** [comment]
I think the `ending` routine needs additional callback when using optional test option. Otherwise users get two similar hints:
```
Your project is now created, you can use the following commands to get going
    cd "WebApplicationBasic/src/WebApplicationBasic"
...
Your project is now created, you can use the following commands to get going
    cd "WebApplicationBasic.Tests"
```
**And the hint:**
```
 cd "WebApplicationBasic.Tests"
```
just yelds:
```
cd "WebApplicationBasic.Tests"
cd: no such file or directory: WebApplicationBasic.Tests
```

**Comment by peterblazejewicz** [comment]
@sayedihashimi @spboyer @Spinarooni @scottaddie 
Any thoughts? The rc-1 is nearing and the PR will diverge significantly if left.
The points I've raised can be easily fixed/changed later via separate, new issues.
Thanks!

File: app/index.js (hunk 1 above)
Comment (david-driscoll) [inline app/index.js]: Honestly doesn't matter to me, the naming of "file" probably needs to change.  Maybe it should just be `Create global.json file?`

**Comment by david-driscoll** [comment]
Sorry got distracted with user groups / code camp / conferences.

I'll see if I can fixup the ending path when you have a test project added, and I'll rebase onto whatever is master at the moment.

**Comment by peterblazejewicz** [comment]
@david-driscoll NP

**Comment by david-driscoll** [comment]
Changed made and rebased, want me to squash as well?

**Comment by shrayasr** [comment]
RC1 has dropped. This would be a really nice to have feature. Right now I create a scaffolding with `yo` and then create the solution around it with VS. Both being done in the same step would be a wonderful thing to have :+1: 

Thank you in advance for your contributions. They really ease the process of development. :beers:

**Comment by peterblazejewicz** [comment]
@david-driscoll 
If you rebase from master there will be only 1 simple conflict (`app/index.js` - at least when I test this now). We could make it live once @sayedihashimi is back from holiday.
Thanks!

**Comment by sayedihashimi** [comment]
@peterblazejewicz @david-driscoll I'm back from holiday. I was thinking that we should release in the next day or so as we have some goodness in the master branch. We can release whenever you guys lile.

**Pull request closed without merging**